osltweaks.php

```diff
@@ -44,6 +44,40 @@ function osltweaks_civicrm_buildForm($formName, &$form) {
         }
       }
     }
+
+    // Get current contact id
+    $currentContactId = CRM_Core_Session::singleton()->getLoggedInContactID();
+    // Get Contributions base on the contribution page id and current contact id
+    $contributions = civicrm_api3('Contribution', 'get', [
```

Comment:
@lucky091588 This is calling the contribution.get api. But the requirement is to check for an existing _membership_ for the user. We don't need to check contributions at all. Instead, please just check to see if $currentContactId has a membership record (of any status).

Reply:
@twomice I think the contribution API is needed since we can't indicate the memberships API that it has the right contribution page. To indicate that the contribution page and membership matches, we need to add the contribution_source on the memberships API. If it doesn't have that, even if the contribution page is not the right page of that membership, but it still has matches the $settings['member_redirect_contribution_pages']. It will still redirect. Let me know your thoughts on this.

Reply:
@lucky091588 Actually we don't care how they got the membership: maybe it was added manually by staff; maybe they signed up through some contribution page that no longer exists; we don't care about that. The only thing we care about is: "do they have a membership record?". Considering this requirement, we only need the membership API, right?

Reply:
@twomice I see. So, no matter what contribution page a customer access, as long as he has membership, and it it already expires. It should redirect right?

Reply:
Yep, exactly. Thanks!